pkg/syncer/statussyncer.go

```diff
@@ -64,6 +71,59 @@ func NewStatusSyncer(from, to *rest.Config, gvrs []string, kcpClusterName logica
 	return New(kcpClusterName, pclusterID, fromClient, toClient, SyncUp, gvrs, pclusterID, mutatorsMap)
 }
 
+func (c *Controller) deleteFromUpstream(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace, name string) error {
+	return c.removeUpstreamSyncerOwnership(ctx, gvr, upstreamNamespace, name)
+}
+
+func (c *Controller) removeUpstreamSyncerOwnership(ctx context.Context, gvr schema.GroupVersionResource, upstreamNamespace, name string) error {
```

Comment:
can we rename to ensureUpstreamFinalizerRemoved ?

Reply:
The changes done to this method based on [this comment](https://github.com/kcp-dev/kcp/pull/942#discussion_r861610345) don't seem right:
- the `upstreamClient dynamic.Interface` argument has been removed, and in replaced in the method by calling `c.toClient`
- However this method is also called, in a corner-case, [in the specsyncer](https://github.com/kcp-dev/kcp/pull/942/files#diff-e03cd260985652e7bb1a27c1c23905e46d846815d4084dd87145f55860c619ddR288) when the downstream object should be deleted, and in fact it has already been deleted.
- So when this method is called from the specsyncer, using the `c.toClient` corresponds to using to client to the downstream kube cluster, while we want this method to act n the upstream KCP logical cluster *in all cases*

To conclude I'd rather make this method a helper function, not a method of the controller, by adding the following parameters to it:
- `upstreamClient dynamic.Interface` (which has been dropped)
- `pcluster string` (take the opportunity to rename it to `workloadCluster`)
- `upstreamClusterName string` (take the opportunity to rename it as `logicalClusterName`)